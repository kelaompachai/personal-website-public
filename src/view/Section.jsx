import React from 'react';
import Intro from './Intro';
import Projects from './Projects';
import Reach from './Reach';
import Recs from './Recs';
import Talks from './Talks';

function Section(props) {
  const { title } = props;
  const fill = title === 'Intro'
    ? <Intro /> : title === 'Open Source Projects'
      ? <Projects /> : title === 'Talks & Publications'
        ? <Talks /> : title === 'Recommendations'
          ? <Recs /> : <Reach />;

  return (
    <section className="min-h-12 border-2 border-black border-solid flex flex-col rounded-3xl mx-32 my-5">
      <div className="flex justify-center mb-5">
        { title !== 'Intro' ? <h3 className="text-center text-2xl py-5 bg bg-well-read-400 rounded-3xl w-1/2">{title}</h3> : null }
      </div>
      { fill }
    </section>
  );
}

export default Section;
