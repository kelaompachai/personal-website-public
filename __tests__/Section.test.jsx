import React from 'react';
import { render, screen } from '@testing-library/react';
import Section from '../src/view/Section';

describe('Section component renders properly', () => {
  it('looks good as Intro', () => {
    const container = render(<Section title="Intro" />);
    expect(container.baseElement).toMatchSnapshot();
    expect(screen.queryByText(/Intro/i)).not.toBeInTheDocument();
  });
  it('looks good as something else', () => {
    const container = render(<Section title="Open Source Projects" />);
    expect(container.baseElement).toMatchSnapshot();
    expect(screen.queryByText(/Open Source Projects/)).toBeInTheDocument();
  });
});
