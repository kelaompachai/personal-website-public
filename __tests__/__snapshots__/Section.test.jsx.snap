// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders properly looks good as Intro 1`] = `
<body>
  <div>
    <section
      class="min-h-12 border-2 border-black border-solid flex flex-col rounded-3xl mx-32 my-5"
    >
      <div
        class="flex justify-center mb-5"
      />
      <div
        class="flex text-lg font-light"
      >
        <img
          alt="My beautiful face"
          class="rounded-full mx-7 object-contain size-36"
          src="testFileStub"
        />
        <p
          class="mx-7"
        >
          Hi, I'm Joe Paul, a software engineer who's passionate about what's true. I enjoy building websites (front and back end), solving leetcode problems, reading classics, and hiking. I'm currently working on this website, learning gradle, and reading Every Valley by Charles King. Feel free to email me if you'd like to talk about any of those things.
        </p>
        <div
          class="flex flex-col items-center"
        >
          <span>
            jobs.alarm618@passinbox.com
          </span>
          <a
            class="text-well-read-400 underline"
            href="https://www.github.com/kelaompachai"
          >
            Github
          </a>
          <a
            class="text-well-read-400 underline"
            href="https://www.linkedin.com/in/joseph-lawrence-paul"
          >
            LinkedIn
          </a>
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`Section component renders properly looks good as something else 1`] = `
<body>
  <div>
    <section
      class="min-h-12 border-2 border-black border-solid flex flex-col rounded-3xl mx-32 my-5"
    >
      <div
        class="flex justify-center mb-5"
      >
        <h3
          class="text-center text-2xl py-5 bg bg-well-read-400 rounded-3xl w-1/2"
        >
          Open Source Projects
        </h3>
      </div>
      <div
        class="grid grid-cols-2 gap-4"
      >
        <div
          class="flex flex-col items-center"
        >
          <h1
            class="text-4xl text-wrap text-white max-w-96 text-center font-extralight"
          >
            The Floor Plan
          </h1>
          <div
            class="min-h-72 flex justify-around mt-5"
          >
            <a
              class="flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover bg-floorplan hover:opacity-80 active:transform active:translate-y-0.5"
              href="https://github.com/kelaompachai/The-Floor-Plan"
            />
          </div>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h1
            class="text-4xl text-wrap text-white max-w-96 text-center font-extralight"
          >
            Pass the Plate
          </h1>
          <div
            class="min-h-72 flex justify-around mt-5"
          >
            <a
              class="flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover bg-passtheplate hover:opacity-80 active:transform active:translate-y-0.5"
              href="https://github.com/kelaompachai/Pass-the-Plate"
            />
          </div>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h1
            class="text-4xl text-wrap text-white max-w-96 text-center font-extralight"
          >
            KafOpticon
          </h1>
          <div
            class="min-h-72 flex justify-around mt-5"
          >
            <a
              class="flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover bg-kafopticon hover:opacity-80 active:transform active:translate-y-0.5"
              href="https://github.com/kelaompachai/KafOpticon"
            />
          </div>
        </div>
        <div
          class="flex flex-col items-center"
        >
          <h1
            class="text-4xl text-wrap text-white max-w-96 text-center font-extralight"
          >
            AlgoGoggles
          </h1>
          <div
            class="min-h-72 flex justify-around mt-5"
          >
            <a
              class="flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover bg-algogoggles hover:opacity-80 active:transform active:translate-y-0.5"
              href="https://github.com/kelaompachai/AlgoGoggles"
            />
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
